src/inabox/inabox-viewport.js

```diff
@@ -81,11 +85,21 @@ export class ViewportBindingInabox {
     // 2) broadcast the request
     this.iframeClient_.setHostWindow(win.top);
 
+    /** @private {boolean} */
+    this.visibilityV2Enabled_ =
+        nativeIntersectionObserverSupported(win) &&
+            isExperimentOn(win, 'visibility-v2');
+
     dev().fine(TAG, 'initialized inabox viewport');
   }
 
   /** @override */
   connect() {
+    if (this.visibilityV2Enabled_) {
+      // Visibility V2 uses native IntersectionObserver, no position data needed
```

Comment:
I wonder does the native IntersectionObserver callback function handle remeasure correctly? Could you link me the code where the callback function is?